cluster_describer.go

```diff
@@ -0,0 +1,65 @@
+package kafka
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/kafka"
+	"github.com/aws/aws-sdk-go-v2/service/kafka/types"
+)
+
+const (
+	clusterConfigTimeout      = 5 * time.Second
+	clusterDescriptionTimeout = 10 * time.Second
+)
+
+type clusterDescriber interface {
+	DescribeClusterV2(ctx context.Context, input *kafka.DescribeClusterV2Input, optFns ...func(*kafka.Options)) (*kafka.DescribeClusterV2Output, error)
+}
+
+func newClusterDescriber() (clusterDescriber, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), clusterConfigTimeout)
+	defer cancel()
+
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return kafka.NewFromConfig(cfg), nil
+}
+
+// Verifies whether the Kafka cluster is available or not.
+// False positive healthcheck errors are being ignored during maintenance windows.
+func checkClusterAvailability(healthErr error, describer clusterDescriber, arn *string) error {
```

Comment:
Definitely not satisfied with this but just copy-pasting the body of the function across all healthchecks is not cool either.

Modifying the error _seems_ to be the correct way to avoid the repetition but I'm open to suggestions if I've missed something obvious.

Reply:
**nitpick (non-blocking):** Just semantics - from a function named *checkClusterAvailability* I wouldn't expect to receive an error as argument. I would expect to check whether the cluster is available or not and to let the code using the function decides what to do in case of Maintenance error. The downside is what you've described. I would rename the function to something like `verifyHealthErrorServerity`.